### PR TITLE
[global] add cloud-provider-zvirt module

### PR DESCRIPTION
**Pull request by aleksey-su**
## Description
Added support of new cloud provider - zVirt

## Why do we need it, and what problem does it solve?
Our clients want to use zVirt

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: Added support of new cloud provider - zVirt.
impact_level: default
```

### CODE REVIEW

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      kubectl -n d8-system exec -ti deploy/deckhouse -- deckhouse-controller edit provider-cluster-configuration
+      ```
+    x-doc-search: |
+      ProviderClusterConfiguration
+    x-unsafe-rules: [deleteZones]
+    x-examples:
+      - apiVersion: deckhouse.io/v1
+        kind: ZvirtClusterConfiguration
+        layout: Standard
+        clusterId: "b46372e7-0d52-40c7-9bbf-fda31e187088"
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:24]: clusterID

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            replicas: 1
+            instanceClass:
+              numCPUs: 4
+              memory: 8192
+              template: debian-bookworm
+        provider:
+          server: '<SERVER>'
+          username: '<USERNAME>'
+          password: '<PASSWORD>'
+          insecure: true
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:45]: ca

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+              memory:
+                type: integer
+                description: |
+                  Memory in MiB to allocate to the VM.
+                example: 8192
+              rootDiskSizeGb:
+                description: |
+                  Root disk size in GiB to use in zVirt VirtualMachines.
+                example: 20
+                default: 20
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:98]: 30

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            x-unsafe-rules: [updateReplicas]
+          instanceClass:
+            type: object
+            additionalProperties: false
+            required: [numCPUs, memory, template]
+            description: |
+              Partial contents of the fields of the [ZvirtInstanceClass](https://deckhouse.ru/documentation/v1/modules/030-cloud-provider-zvirt/cr.html#zvirtinstanceclass).
+            properties: &instanceClassProperties
+              numCPUs:
+                type: integer
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:85]: min 1

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                description: |
+                  Number of vCPUs to allocate to the VM.
+                example: 2
+              memory:
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:89]: min 1

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+                example: debian-bookworm
+              vnicProfileId:
+                type: string
+                description: |
+                  vNIC profile id.
+                example: "49bb4594-0cd4-4eb7-8288-8594eafd5a86"
+              nicName:
+                type: string
+                description: |
+                  vNIC name.
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:119]: add full description

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+                example: "nic1"
+              storageDomainId:
+                type: string
+                description: |
+                  Storage domain id.
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:124]: add full description

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+                  Etcd disk size in gigabytes.
+                example: 10
+                default: 10
+                type: integer
+              template:
+                type: string
+                description: |
+                  Template name to be cloned.
+                example: debian-bookworm
+              vnicProfileId:
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:111]: required

File: ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+              nicName:
+                type: string
+                description: |
+                  vNIC name.
+                example: "nic1"
+              storageDomainId:
+                type: string
+                description: |
+                  Storage domain id.
+                example: "49bb4594-0cd4-4eb7-8288-8594eafd5a86"
```
Comment (aleksey-su) [inline ee/candi/cloud-providers/zvirt/openapi/cluster_configuration.yaml:125]: regexp uuid

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8108](https://github.com/deckhouse/deckhouse/pull/8108) to the branch [release-1.59](https://github.com/deckhouse/deckhouse/tree/release-1.59) successful!

**Pull request merged**